dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java

```diff
@@ -182,6 +182,10 @@ else if ( !nonAggDataElements.isEmpty() )
         {
             error = new ErrorMessage( ErrorCode.E7115, getUids( nonAggDataElements ) );
         }
+        else if ( !params.getSkipTotalDataElements().isEmpty() )
```

Comment:
The code looks good @larshelge :+1:, my only concern would be this validation.
I'm afraid that requests that used to work start failing, and users may start asking that they would like to continue seeing those values. I'm thinking about the situation where some dashboards or visualizations will start "failing" from the user's point of view.

Reply:
I agree. Though in this case, if users have set "skip total" on the category combination, that must be for a reason, which is that the total is not meaningful and should not be displayed in reports/visualizations. So I think this is making the solution more correct. Also, having the "skip total" setting enabled is quite rare from what I have seen.